spec/shadow/index.html

```diff
@@ -1149,21 +1149,150 @@
       </section>
 
       <section>
-        <h3>Focus Navigation</h3>
+        <h3>Sequential Focus Navigation</h3>
 
-        <p>If a <a>node</a> doesn’t <a data-lt="participates">participate</a> in the <a>document flat tree</a>, the <a>node</a> <strong>must</strong> be skipped from the <a>sequential focus navigation</a>.</p>
+        <p><dfn>focus navigation scope</dfn> is a set of elements <a>in a composed document</a>. An element belongs to at most one <a>focus navigation scope</a>.</p>
 
-        <p>The <a>sequential focus navigation order</a> for a given <a>shadow tree</a> <var>A</var> <strong>must</strong> be inserted into the <a>sequential focus navigation order</a> for the <a>parent tree</a> <var>B</var> as follows:</p>
-        <ol>
-          <li>Let <var>HOST</var> be the <a>shadow host</a> which <a>hosts</a> A</li>
-          <li>
-            The <a>sequential focus navigation order</a> for A <strong>must</strong> be inserted into the <a>sequential focus navigation order</a> for <var>B</var>:
-            <ol>
-              <li>immediately after <var>HOST</var>, if <var>HOST</var> is <a>focusable</a>; or</li>
-              <li>in place of the <var>HOST</var> as if <var>HOST</var> were assigned the <a>tabindex</a> value 0 for determining its position.</li>
-            </ol>
-          </li>
-        </ol>
+        <p><dfn>focus navigation scope owner</dfn> is a node that owns a <a>focus navigation scope</a>. A document, a <a>shadow root</a>, or a <a>slot element</a> can be a <a>focus navigation scope owner</a>.</p>
+
+        <p><dfn>document sequential focus navigation order</dfn> is an order of all of the <a data-lt="focusable area">focusable areas</a> <a>in a composed document</a>, reachable by <a>sequential focus navigation</a>.</p>
+
+        <p>When a composed document is given, the <a>document sequential focus navigation order</a> is determined by the following steps:</p>
+
+        <p>Step 1.</p>
+        <div class="algorithm">
+          <dl>
+            <dt>INPUT</dt>
+            <dd><var>DOCUMENT</var>, a composed document</dd>
+            <dt>OUTPUT</dt>
+            <dd><var>OWNERS</var>, a set of <a>focus navigation scope owner</a> nodes</dd>
+          </dl>
+          <ol>
+            <li>Let <var>OWNERS</var> be an empty set.</li>
+            <li>For each node tree <var>TREE</var> in <var>DOCUMENT</var>:
```

Comment:
Now you can use https://dom.spec.whatwg.org/#concept-shadow-including-tree-order.
Or, given that the order is not important, you can simply say:
- Let OWNERS be focus owner nodes whose shadow-including root is *document*.